test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import "mocha";
+import * as chai from "chai";
+const should = chai.should();
+import * as chaiAsPromised from "chai-as-promised";
+import * as dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+import { Namespace, SubscriptionClient } from "../lib";
+
+// We need to remove rules before adding one because otherwise the existing default rule will let in all messages.
+async function removeAllRules(client: SubscriptionClient): Promise<void> {
+  try {
+    const rules = await client.getRules();
+    for (let i = 0; i < rules.length; i++) {
+      const rule = rules[i];
+      await client.removeRule(rule.name);
+    }
+  } catch (err) {
+    console.log("Error while removing the rule", err);
+  }
+}
+
+let namespace: Namespace;
+let subscriptionClient: SubscriptionClient;
+let defaultSubscriptionClient: SubscriptionClient;
+
+async function beforeEachTest(): Promise<void> {
+  // The tests in this file expect the env variables to contain the connection string and
+  // the names of empty queue/topic/subscription that are to be tested
+
+  if (!process.env.SERVICEBUS_CONNECTION_STRING) {
+    throw new Error(
+      "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+    );
+  }
+  if (!process.env.TOPIC_NAME) {
+    throw new Error("Define TOPIC_NAME in your environment before running integration tests.");
+  }
+  if (!process.env.SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+  if (!process.env.DEFAULT_SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define DEFAULT_SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+
+  namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+  subscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.SUBSCRIPTION_NAME
+  );
+  defaultSubscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.DEFAULT_SUBSCRIPTION_NAME
+  );
+  await removeAllRules(subscriptionClient);
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+describe("Topic Filters - Tests", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  it("Add Rule with Boolean filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("TrueFilter", true);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "TrueFilter");
```

Comment:
In every test where a rule is being added, check for the length of the `rules` array